tests/test_attribute_definition.py

```diff
@@ -268,7 +268,8 @@ def test_annotated_constraint_fail():
     """Test that the verifier of an annotated constraint can fail."""
     with pytest.raises(Exception) as e:
         PositiveIntAttr([IntData(-42)])
-    assert e.value.args[0] == "Expected positive integer, got -42."
+    assert e.value.args[
+        0] == "The following constraints were not satisfied:\nExpected positive integer, got -42."
```

Comment:
I think it is too verbose?

Reply:
I think Mathieu [suggested this](https://github.com/xdslproject/xdsl/pull/467#discussion_r1116415009) for cases where multiple constraints fail, so as to let users know that we capture failure data for all of them and present that below. 

So I agree that this might seem a bit too verbose for cases with a single constraint, but it should be handy for multiple constraint failures.

Reply:
Yeah, we can probably have a special case for single case failures, where we don't add more text.
Otherwise, I wouldn't say that this is too verbose when multiple constraints fails. These are errors you should only get if your compiler is wrong, so it is fine to me to have verbose errors.